### PR TITLE
Fix agg spilt bug

### LINKED ISSUE #6993: Size of WeakHash32 does not match size of column
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

clickbench

```mysql
mysql> SELECT UserID, SearchPhrase, COUNT(*) FROM hits GROUP BY UserID, SearchPhrase LIMIT 10;
ERROR 1105 (HY000): other error for mpp stream: From MPP<query:<query_ts:1678264801169881664, local_query_id:62, server_id:2818397, start_ts:439947048025587714>,task_id:1>: Code: 49, e.displayText() = DB::Exception: Size of WeakHash32 does not match size of column: column size is 0, hash size is 28144, e.what() = DB::Exception,
```

### 2. What did you expect to see? (Required)

success 

### 3. What did you see instead (Required)

fail

### 4. What is your TiFlash version? (Required)
v6.6.0

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #6993

Problem Summary:

### What is changed and how it works?
The root cause is `convertToBlocksImplFinal` and `convertToBlocksImplNotFinal` does not handle the key column correctly when the group by key has some special optimaztion.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/Aggregator.cpp [hunk 1]
```diff
@@ -1263,19 +1277,33 @@ template <typename Method, typename Table>
...
+    for (size_t i = 1; i < key_columns_vec.size(); i++)
+    {
+        auto new_key_sizes = method.shuffleKeyColumns(key_columns_vec[i], key_sizes);
+        assert(shuffled_key_sizes == new_key_sizes);
+    }
     const auto & key_sizes_ref = shuffled_key_sizes ? *shuffled_key_sizes : key_sizes;
 
-    AggregatorMethodInitKeyColumnHelper<Method> agg_keys_helper{method};
-    agg_keys_helper.initAggKeys(data.size(), key_columns_vec[0]);
+    std::vector<std::unique_ptr<AggregatorMethodInitKeyColumnHelper<Method>>> agg_keys_helpers;
```
Comment (yibin87) [inline dbms/src/Interpreters/Aggregator.cpp:1291]: Extract a common method for usage for L1291-L1300?

**Review by yibin87 — APPROVED** [review]
Others LGTM

File: dbms/src/Interpreters/Aggregator.cpp (hunk 1 above)
Comment (windtalker) [reply dbms/src/Interpreters/Aggregator.cpp]: Done.

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**